### DISCUSSION TITLE
[Feature] Legacy shared power

### DISCUSSION THREAD

**Discussion by MUGEN-Slayer** [ikemen-engine/Ikemen-GO#855] [Feature Request]
Some time ago I started converting one of my mugen lifebars to IKEMEN and saw that it was possible to play simul mode without having to share the powerbar with the teammate. So I decided to make a powerbar for each member of the team.
![ikemen000](https://user-images.githubusercontent.com/68881657/104223610-9fb51c80-5422-11eb-9a56-ef3746a7275c.png)

However that creates a particular problem... Let's say that after I release the lifebar, there will be people who will play with the shared power. In this case the design will be broken:
![ikemen001](https://user-images.githubusercontent.com/68881657/104223085-ef471880-5421-11eb-9b0b-afd11ec5e59b.png)

It's even more evident in a 4v4 battle:
![ikemen002](https://user-images.githubusercontent.com/68881657/104223117-fbcb7100-5421-11eb-8f10-10f076fca348.png)

In this specific set, the ideal would be that if the user will use the team power share option activated, it would revert to the model used in single mode:
![ikemen003](https://user-images.githubusercontent.com/68881657/104223207-1a316c80-5422-11eb-80a0-154f78415234.png)

Having the following result:
![ikemen004](https://user-images.githubusercontent.com/68881657/104223318-3cc38580-5422-11eb-882b-83e9381646db.png)

My suggestion is to add the parameter "legacysharedpower" to the [Info] section of the lifebars.
If:
- legacysharedpower = 1

When team power share is activated, the powerbars in simul and tag modes will be the same as those used in single mode;

If:
- legacysharedpower = 0

Then the powerbars will be read in the same way as they are currently, each mode having its respective powerbars.
If the parameter is omitted, then legacysharedpower = 1.

In my opinion something like this is important because of the infinite possibilities of designs, allow me to use this mock-up with the Bleach Dark Souls Lifebars (although I have converted them to MUGEN, I have not yet started its migration to IKEMEN).

1vs1
![bds1](https://user-images.githubusercontent.com/68881657/104224993-87460180-5424-11eb-805e-b69f27d1c1ca.png)

"2vs2", power share enabled
![bds2](https://user-images.githubusercontent.com/68881657/104225007-8a40f200-5424-11eb-8292-d9a65025a592.png)

"2vs2", power share disabled
![bds3](https://user-images.githubusercontent.com/68881657/104225032-9331c380-5424-11eb-8f78-9451333db420.png)

Hypothetical case if legacysharedpower = 1
![bds4](https://user-images.githubusercontent.com/68881657/104225038-975de100-5424-11eb-87d3-83e38b0533cb.png)

PS. Please ignore the timer.

In the current state of the engine, some lifebars are going to have serious design problems as in the first case and others not, as in the second. I believe that such function would resolve this impasse. I suppose the same would apply to team life share.

Taking advantage of the theme, how about having instead:
- "team life share" and "team power share"

we have this:
- "simul life share"/" simul power share" and "tag life share"/"tag power share"

I think it would be better in simul mode for the powerbar not to be shared, while in tag mode the team uses it shared. In the current model it is not possible, and this would allow complete freedom by default.

**Comment by lenocid** [comment]
Now this feature still doesn't realize.
